src/components/app.tsx

```diff
@@ -285,11 +306,18 @@ export default function App(props: AppProps): JSX.Element {
     setLiftSpotlight(undefined);
     setRobotSpotlight(undefined);
     setDispenserSpotlight(undefined);
+    setNegotiationSpotlight(undefined);
   }
 
+  function removeAllNegotiationTrajectories(): void {
+    for (var key of Object.keys(negotiationTrajStore))
+      delete negotiationTrajStore[key];
+  }
```

Comment:
If I'm understanding right and the objective it's to remove all negotiations, you can do the following:
```
const [negotiationTrajStore, setNegotiationTrajStore] = React.useState
    <Record<string, NegotiationTrajectoryResponse>>({});
```
on line 165 and 
you can use 
`setNegotiationTrajStore({})` to remove them all.


When you're doing this 
```
const [negotiationTrajStore, setNegotiationTrajStore] = React.useState
    <Record<string, NegotiationTrajectoryResponse>>({});
```
You're defining a state that it'll be a store here `negotiationTrajStore` and the store modifier which is this `setNegotiationTrajStore`, in theory, you should modify the state only with its state modifier.

HTH.

Reply:
replace function with `setNegotiationTrajStore({});` 6fb5321